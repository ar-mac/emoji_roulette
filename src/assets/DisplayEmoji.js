import React, { Component } from 'react';
import twemoji from 'twemoji'
import PropTypes from 'proptypes';

import { Emojis } from './emojis.js';

class DisplayEmoji extends Component {
  state = {
    secondsPassed: 0,
  };

  componentDidMount() {
    this.intervalId = setInterval(() => {
      this.setState({ secondsPassed: this.state.secondsPassed + 1 })
    }, 1000);
  }

  componentDidUpdate() {
    if (this.state.secondsPassed === 2) {
      this.props.getNewEmoji();
    }
  }

  componentWillUnmount() {
    clearInterval(this.intervalId);
  }

  render() {
    const { index, resetFn } = this.props;
    const selectedEmoji = Emojis[index];
    const twemojiMarkup = {
      __html: twemoji.parse(selectedEmoji.emoji)
    };

    return (
      <div className="test">
        <div>secondsPassed: {this.state.secondsPassed}</div>
        <button onClick={resetFn}>Reset index</button>
        <div>emoji: {selectedEmoji.emoji}</div>
        <div>codepoint: {selectedEmoji.codepoint}</div>
        <div>parsed codepoint: {String.fromCodePoint(selectedEmoji.codepoint)}</div>
        <div>twemoji: <div dangerouslySetInnerHTML={twemojiMarkup}/></div>
        <div>index: {index}</div>
      </div>
    )
  }
}

DisplayEmoji.propTypes = {
  index: PropTypes.number.isRequired,
};

export default DisplayEmoji
