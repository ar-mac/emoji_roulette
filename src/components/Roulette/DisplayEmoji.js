import React, { Component } from 'react';
import twemoji from 'twemoji'
import PropTypes from 'proptypes';

import { withTimer } from './withTimer';

export class DisplayEmoji extends Component {
  render() {
    const { index, clearIndex, secondsPassed, emoji } = this.props;
    const twemojiMarkup = {
      __html: twemoji.parse(emoji.emojiIcon)
    };

    return (
      <div className="emoji-card">
        <div className="emoji-card__head">
          {secondsPassed >= 0 && <div>secondsPassed: {secondsPassed}</div>}
          {!!clearIndex && <div><button onClick={clearIndex}>Clear index</button></div>}
        </div>
        <div className="emoji-card__body">
          <div>emoji: {emoji.emojiIcon}</div>
          <div>code: {emoji.codepoint}</div>
          <div>parsed: {String.fromCodePoint(emoji.codepoint)}</div>
          <div>twemoji: <div dangerouslySetInnerHTML={twemojiMarkup}/></div>
          <div>index: {index}</div>
        </div>
      </div>
    )
  }
}

DisplayEmoji.propTypes = {
  index: PropTypes.number.isRequired,
};

export default withTimer(DisplayEmoji);
