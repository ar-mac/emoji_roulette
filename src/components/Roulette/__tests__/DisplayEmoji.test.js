import React from 'react';
import {DisplayEmoji} from '../DisplayEmoji';

jest.mock('twemoji', () => ({
  parse: (icon) => `some mocked string ${icon}`,
}));

describe('<DisplayEmoji />', () => {
  const defaultProps = {
    index: 0,
    clearIndex: () => {},
    secondsPassed: 0,
    emoji: { codepoint: 0x1F600, emojiIcon: 'ðŸ˜€' },
  };

  const setup = buildSetup(DisplayEmoji, defaultProps);

  test('renders with default props', () => {
    const { wrapper } = setup();
    expect(wrapper.instance()).toMatchSnapshot();
  });

  test('renders with invalid prop', () => {
    const { wrapper } = setup({
      secondsPassed: undefined,
    });
    expect(wrapper.instance()).toMatchSnapshot();
  });
  
  test('renders without clearIndex', () => {
    const { wrapper } = setup({clearIndex: undefined });
    expect(wrapper.instance()).toMatchSnapshot();

  });
  
  test('triggers function when click clearIndex button', () => {
    const { wrapper } = setup({clearIndex: jest.fn()});
    wrapper.find('button').simulate('click');
    expect(wrapper.instance().props.clearIndex).toHaveBeenCalled();
  });
  
});


