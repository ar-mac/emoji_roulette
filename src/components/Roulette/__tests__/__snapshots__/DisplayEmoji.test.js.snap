// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayEmoji /> renders with default props 1`] = `
DisplayEmoji {
  "_reactInternalInstance": ShallowComponentWrapper {
    "_calledComponentWillUnmount": false,
    "_compositeType": 0,
    "_context": Object {},
    "_currentElement": <DisplayEmoji
      clearIndex={[Function]}
      emoji={
            Object {
                  "codepoint": 128512,
                  "emojiIcon": "ðŸ˜€",
                }
      }
      index={0}
      secondsPassed={0}
/>,
    "_debugID": 1,
    "_hostContainerInfo": null,
    "_hostParent": null,
    "_instance": [Circular],
    "_mountOrder": 1,
    "_pendingCallbacks": null,
    "_pendingElement": null,
    "_pendingForceUpdate": false,
    "_pendingReplaceState": false,
    "_pendingStateQueue": null,
    "_renderedComponent": NoopInternalComponent {
      "_currentElement": <div
        className="emoji-card"
>
        <div
                className="emoji-card__head"
        >
                <div>
                        secondsPassed: 
                        0
                </div>
                <div>
                        <button
                                onClick={[Function]}
                        >
                                Clear index
                        </button>
                </div>
        </div>
        <div
                className="emoji-card__body"
        >
                <div>
                        emoji: 
                        ðŸ˜€
                </div>
                <div>
                        code: 
                        128512
                </div>
                <div>
                        parsed: 
                        ðŸ˜€
                </div>
                <div>
                        twemoji: 
                        <div
                                dangerouslySetInnerHTML={
                                        Object {
                                                "__html": "some mocked string ðŸ˜€",
                                              }
                                }
                        />
                </div>
                <div>
                        index: 
                        0
                </div>
        </div>
</div>,
      "_debugID": 2,
      "_renderedOutput": <div
        className="emoji-card"
>
        <div
                className="emoji-card__head"
        >
                <div>
                        secondsPassed: 
                        0
                </div>
                <div>
                        <button
                                onClick={[Function]}
                        >
                                Clear index
                        </button>
                </div>
        </div>
        <div
                className="emoji-card__body"
        >
                <div>
                        emoji: 
                        ðŸ˜€
                </div>
                <div>
                        code: 
                        128512
                </div>
                <div>
                        parsed: 
                        ðŸ˜€
                </div>
                <div>
                        twemoji: 
                        <div
                                dangerouslySetInnerHTML={
                                        Object {
                                                "__html": "some mocked string ðŸ˜€",
                                              }
                                }
                        />
                </div>
                <div>
                        index: 
                        0
                </div>
        </div>
</div>,
    },
    "_renderedNodeType": 0,
    "_rootNodeID": 0,
    "_topLevelWrapper": null,
    "_updateBatchNumber": null,
    "_warnedAboutRefsInRender": false,
  },
  "context": Object {},
  "props": Object {
    "clearIndex": [Function],
    "emoji": Object {
      "codepoint": 128512,
      "emojiIcon": "ðŸ˜€",
    },
    "index": 0,
    "secondsPassed": 0,
  },
  "refs": Object {},
  "state": null,
  "updater": Object {
    "enqueueCallback": [Function],
    "enqueueCallbackInternal": [Function],
    "enqueueElementInternal": [Function],
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
    "validateCallback": [Function],
  },
}
`;

exports[`<DisplayEmoji /> renders with invalid prop 1`] = `
DisplayEmoji {
  "_reactInternalInstance": ShallowComponentWrapper {
    "_calledComponentWillUnmount": false,
    "_compositeType": 0,
    "_context": Object {},
    "_currentElement": <DisplayEmoji
      clearIndex={[Function]}
      emoji={
            Object {
                  "codepoint": 128512,
                  "emojiIcon": "ðŸ˜€",
                }
      }
      index={0}
      secondsPassed={undefined}
/>,
    "_debugID": 3,
    "_hostContainerInfo": null,
    "_hostParent": null,
    "_instance": [Circular],
    "_mountOrder": 2,
    "_pendingCallbacks": null,
    "_pendingElement": null,
    "_pendingForceUpdate": false,
    "_pendingReplaceState": false,
    "_pendingStateQueue": null,
    "_renderedComponent": NoopInternalComponent {
      "_currentElement": <div
        className="emoji-card"
>
        <div
                className="emoji-card__head"
        >
                <div>
                        <button
                                onClick={[Function]}
                        >
                                Clear index
                        </button>
                </div>
        </div>
        <div
                className="emoji-card__body"
        >
                <div>
                        emoji: 
                        ðŸ˜€
                </div>
                <div>
                        code: 
                        128512
                </div>
                <div>
                        parsed: 
                        ðŸ˜€
                </div>
                <div>
                        twemoji: 
                        <div
                                dangerouslySetInnerHTML={
                                        Object {
                                                "__html": "some mocked string ðŸ˜€",
                                              }
                                }
                        />
                </div>
                <div>
                        index: 
                        0
                </div>
        </div>
</div>,
      "_debugID": 4,
      "_renderedOutput": <div
        className="emoji-card"
>
        <div
                className="emoji-card__head"
        >
                <div>
                        <button
                                onClick={[Function]}
                        >
                                Clear index
                        </button>
                </div>
        </div>
        <div
                className="emoji-card__body"
        >
                <div>
                        emoji: 
                        ðŸ˜€
                </div>
                <div>
                        code: 
                        128512
                </div>
                <div>
                        parsed: 
                        ðŸ˜€
                </div>
                <div>
                        twemoji: 
                        <div
                                dangerouslySetInnerHTML={
                                        Object {
                                                "__html": "some mocked string ðŸ˜€",
                                              }
                                }
                        />
                </div>
                <div>
                        index: 
                        0
                </div>
        </div>
</div>,
    },
    "_renderedNodeType": 0,
    "_rootNodeID": 0,
    "_topLevelWrapper": null,
    "_updateBatchNumber": null,
    "_warnedAboutRefsInRender": false,
  },
  "context": Object {},
  "props": Object {
    "clearIndex": [Function],
    "emoji": Object {
      "codepoint": 128512,
      "emojiIcon": "ðŸ˜€",
    },
    "index": 0,
    "secondsPassed": undefined,
  },
  "refs": Object {},
  "state": null,
  "updater": Object {
    "enqueueCallback": [Function],
    "enqueueCallbackInternal": [Function],
    "enqueueElementInternal": [Function],
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
    "validateCallback": [Function],
  },
}
`;

exports[`<DisplayEmoji /> renders without clearIndex 1`] = `
DisplayEmoji {
  "_reactInternalInstance": ShallowComponentWrapper {
    "_calledComponentWillUnmount": false,
    "_compositeType": 0,
    "_context": Object {},
    "_currentElement": <DisplayEmoji
      clearIndex={undefined}
      emoji={
            Object {
                  "codepoint": 128512,
                  "emojiIcon": "ðŸ˜€",
                }
      }
      index={0}
      secondsPassed={0}
/>,
    "_debugID": 5,
    "_hostContainerInfo": null,
    "_hostParent": null,
    "_instance": [Circular],
    "_mountOrder": 3,
    "_pendingCallbacks": null,
    "_pendingElement": null,
    "_pendingForceUpdate": false,
    "_pendingReplaceState": false,
    "_pendingStateQueue": null,
    "_renderedComponent": NoopInternalComponent {
      "_currentElement": <div
        className="emoji-card"
>
        <div
                className="emoji-card__head"
        >
                <div>
                        secondsPassed: 
                        0
                </div>
        </div>
        <div
                className="emoji-card__body"
        >
                <div>
                        emoji: 
                        ðŸ˜€
                </div>
                <div>
                        code: 
                        128512
                </div>
                <div>
                        parsed: 
                        ðŸ˜€
                </div>
                <div>
                        twemoji: 
                        <div
                                dangerouslySetInnerHTML={
                                        Object {
                                                "__html": "some mocked string ðŸ˜€",
                                              }
                                }
                        />
                </div>
                <div>
                        index: 
                        0
                </div>
        </div>
</div>,
      "_debugID": 6,
      "_renderedOutput": <div
        className="emoji-card"
>
        <div
                className="emoji-card__head"
        >
                <div>
                        secondsPassed: 
                        0
                </div>
        </div>
        <div
                className="emoji-card__body"
        >
                <div>
                        emoji: 
                        ðŸ˜€
                </div>
                <div>
                        code: 
                        128512
                </div>
                <div>
                        parsed: 
                        ðŸ˜€
                </div>
                <div>
                        twemoji: 
                        <div
                                dangerouslySetInnerHTML={
                                        Object {
                                                "__html": "some mocked string ðŸ˜€",
                                              }
                                }
                        />
                </div>
                <div>
                        index: 
                        0
                </div>
        </div>
</div>,
    },
    "_renderedNodeType": 0,
    "_rootNodeID": 0,
    "_topLevelWrapper": null,
    "_updateBatchNumber": null,
    "_warnedAboutRefsInRender": false,
  },
  "context": Object {},
  "props": Object {
    "clearIndex": undefined,
    "emoji": Object {
      "codepoint": 128512,
      "emojiIcon": "ðŸ˜€",
    },
    "index": 0,
    "secondsPassed": 0,
  },
  "refs": Object {},
  "state": null,
  "updater": Object {
    "enqueueCallback": [Function],
    "enqueueCallbackInternal": [Function],
    "enqueueElementInternal": [Function],
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
    "validateCallback": [Function],
  },
}
`;
